components/channel_header_dropdown/index.js

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {createSelector} from 'reselect';
+import {
+    getCurrentUser,
+    getUserStatuses,
+} from 'mattermost-redux/selectors/entities/users';
+import {
+    getCurrentChannel,
+    isCurrentChannelDefault,
+    isCurrentChannelFavorite,
+    isCurrentChannelMuted,
+    isCurrentChannelArchived,
+    isCurrentChannelReadOnly,
+} from 'mattermost-redux/selectors/entities/channels';
+
+import {Constants} from 'utils/constants';
+import * as Utils from 'utils/utils';
+
+import Desktop from './channel_header_dropdown';
+import Mobile from './mobile_channel_header_dropdown';
+
+const getTeammateId = createSelector(
+    getCurrentChannel,
+    (channel) => {
+        if (channel.type !== Constants.DM_CHANNEL) {
+            return null;
+        }
+
+        return Utils.getUserIdFromChannelName(channel);
+    },
+);
+
+const getTeammateStatus = createSelector(
+    getUserStatuses,
+    getTeammateId,
+    (userStatuses, teammateId) => {
+        if (!teammateId) {
+            return null;
+        }
+
+        return userStatuses[teammateId];
+    }
+);
+
+const mapStateToProps = (state) => ({
+    user: getCurrentUser(state),
+    channel: getCurrentChannel(state),
+    isDefault: isCurrentChannelDefault(state),
+    isFavorite: isCurrentChannelFavorite(state),
+    isMuted: isCurrentChannelMuted(state),
+    isReadonly: isCurrentChannelReadOnly(state),
+    isArchived: isCurrentChannelArchived(state),
+    teammateId: getTeammateId(state),
+    teammateStatus: getTeammateStatus(state),
+});
+
+export const ChannelHeaderDropdown = connect(mapStateToProps)(Desktop);
+export const MobileChannelHeaderDropdown = connect(mapStateToProps)(Mobile);
```

Comment:
@saturninoabril that's good, but how about the import-style?

I thought named export would be better for this, users might need some auto-complete, auto-import because of a couple of components in the same index file in this case.

```
import ChannelHeaderDropdown, { MobileChannelHeaderDropdown } from '...'
```

vs

```
import { ChannelHeaderDropdown, MobileChannelHeaderDropdown } from '...'
```

Reply:
Yeah, I think it make sense.  I'm good at what you have right now.